torb/trigger_staging_build/service.py

```diff
@@ -15,55 +16,98 @@ def handler(event, context):
     determined using beanstalk_utils.whodaman, which requires ElasticBeanstalk
     IAM permissions
     '''
-    # determine if we are deploying to fourfront-webprod or fourfront-webprod2
+    # determine if we are deploying to a production env
     event['source_env'] = bs.whodaman()
-    if event['source_env'] == 'fourfront-webprod':
-        event['dest_env'] = 'fourfront-webprod2'
-    elif event['source_env'] == 'fourfront-webprod2':
-        event['dest_env'] = 'fourfront-webprod'
-    else:
-        # event['source_env'] is unexpected. Bail
+    source_env = event['source_env']
+    dest_env = None
+    if is_fourfront_env(source_env) and is_stg_or_prd_env(source_env):
+        # TODO: Making this cgap-compatible means making an adjustment indicated later in this file. -kmp 10-Apr-2020
+        dest_env = guess_mirror_env(source_env)
+    if not dest_env:
+        # e.g., can get here if:
+        #  - source_env is a cgap environment (downstream isn't yet ready for that)
+        #  - source_env is not a production environment
+        #  - source_env is a production environment with no mirror
         return {"type": "trigger_staging_build",
                 "torb_message": "invalid event.source_env", "event": event}
+    event['dest_env'] = dest_env
 
     start_time = datetime.now().strftime("%a_%b_%d_%Y__%H%M%S")
-    run_name = "%s_deploy_for_%s" % (event['dest_env'], start_time)
+    run_name = "%s_deploy_for_%s" % (dest_env, start_time)
     event['start_time'] = start_time
 
     if event.get('run_name'):
         run_name = event.get('run_name')  # used for testing
 
+    try:
+        input = make_input(event)
```

Comment:
Is there a reason for this change? Do you find it fails?

Reply:
I made it fail. See below where I raise that error. There was nothing that would catch an error and I was trying to follow the code flow to see what would happen. I think these coding patterns should be set up with something like this around the outside so that programs can just raise errors and not make up hokey JSON bits at every point. That would be more natural.